### PR TITLE
Add post and delete methods to user-group/id/users to add/remove user…

### PR DESCRIPTION
**Pull request by Migaroez**
### Prerequisites

- [x] I have added steps to test this contribution in the description below.

### Description
After the review of https://github.com/umbraco/Umbraco-CMS/pull/15436 and some extra investigation, we settled on a different approach. This PR is the result of that

Added a way to assign users to usergroups when creating or updating a usergroup. This way you do not need to fetch all the users who have the group, remove them and resave the users.

### Breaking changes
None, all changes to signatures were done on new v14 services/models/controllers

### Testing
Create new users/groups trough the management Api and use the newly created endpoints to add/remove the user(s) to the new and existing group(s).

### CODE REVIEW

File: src/Umbraco.Cms.Api.Management/Controllers/UserGroup/AddUsersToUserGroupController.cs [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+    {
+        IUserGroup? existingUserGroup = await _userGroupService.GetAsync(id);
+
+        if (existingUserGroup is null)
+        {
+            return UserGroupOperationStatusResult(UserGroupOperationStatus.NotFound);
+        }
+
+        UserGroupOperationStatus result = await _userGroupService.AddUsersToUserGroup(
+            new UsersToUserGroupManipulationModel(existingUserGroup, userIds), CurrentUserKey(_backOfficeSecurityAccessor));
```
Comment (bergmania) [inline src/Umbraco.Cms.Api.Management/Controllers/UserGroup/AddUsersToUserGroupController.cs:43]: Any reason this is not a single service call?

File: src/Umbraco.Cms.Api.Management/Controllers/UserGroup/RemoveUsersFromUserGroupController.cs [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+    {
+        IUserGroup? existingUserGroup = await _userGroupService.GetAsync(id);
+
+        if (existingUserGroup is null)
+        {
+            return UserGroupOperationStatusResult(UserGroupOperationStatus.NotFound);
+        }
+
+        UserGroupOperationStatus result = await _userGroupService.RemoveUsersFromUserGroup(
+            new UsersToUserGroupManipulationModel(existingUserGroup, userIds), CurrentUserKey(_backOfficeSecurityAccessor));
```
Comment (bergmania) [inline src/Umbraco.Cms.Api.Management/Controllers/UserGroup/RemoveUsersFromUserGroupController.cs:43]: Any reason this is not a single service call?

File: src/Umbraco.Core/Models/UsersToUserGroupManipulationModel.cs [hunk 3]
```diff
@@ -0,0 +1,15 @@
+using Umbraco.Cms.Core.Models.Membership;
+
+namespace Umbraco.Cms.Core.Models;
+
+public class UsersToUserGroupManipulationModel
+{
+    public IUserGroup UserGroup { get; init; }
```
Comment (bergmania) [inline src/Umbraco.Core/Models/UsersToUserGroupManipulationModel.cs:7]: Could just be the key and handle everything in the service layer

**Review by bergmania — COMMENTED** [review]
A few comments.. Let be know if you agree, then I can just fix it up before I test

**Comment by Migaroez** [comment]
Can't see a reason why not to do it the way you proposed, i just noticed some bad naming too. Ill fix this up

**Review by bergmania — APPROVED** [review]

**Review by bergmania — CHANGES_REQUESTED** [review]
Ill add more auth.. We need to ensure non admins cant add users to the admin user group

**Comment by bergmania** [comment]
> Ill add more auth.. We need to ensure non admins cant add users to the admin user group

I see it was handled in the service too.. But to align with ByKeyUserGroupController I added authentication service check before we call the service.

**Review by bergmania — APPROVED** [review]

**Pull request merged**